modules/core/05-port/types/router_v2.go

```diff
@@ -5,67 +5,109 @@ package types
 import (
 	"errors"
 	"fmt"
-
-	"golang.org/x/exp/maps"
+	"strings"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
-// AppRouter contains a map from module name to an ordered list of IBCModules
-// which contains all the module-defined callbacks required by ICS-26.
+const sentinelMultiPacketData = "MultiPacketData"
+
+// AppRouter contains all the module-defined callbacks required by ICS-26
 type AppRouter struct {
-	routes map[string][]IBCModule
-	sealed bool
+	routes       map[string]IBCModule
+	legacyRoutes map[string]ClassicIBCModule
+
+	// classicRoutes facilitates the consecutive calls to AddRoute for existing modules.
+	classicRoutes map[string][]ClassicIBCModule
```

Comment:
also aware having 2 things called legacy and classic of almost the same type is terrible naming, so we can change that :D

Reply:
maybe we can add a todo that one will be removed?